nexusutils/dataconverter/readers/rii_database/reader.py

```diff
@@ -37,77 +43,307 @@ def read_yml_file(filename: str) -> Dict[Any, Any]:
     return yml_file
 
 
-def read_dispersion(filename: str, identifier: str = 'dispersion_x') -> Dict[str, Any]:
-    """Reads rii dispersions from yaml files"""
+def is_table_k_formula_n_disp(dispersion1: dict, dispersion2: dict):
+    """Checks if the two dispersions are one tabular k formula and one formula n"""
+    if "tabulated k" == dispersion1["type"] and dispersion2["type"].startswith(
+        "formula"
+    ):
+        return True
+
+    if "tabulated k" == dispersion2["type"] and dispersion1["type"].startswith(
+        "formula"
+    ):
+        return True
+
+    return False
+
+
+def read_metadata(yml_file: dict) -> Dict[str, Any]:
+    """Reads metadata from a dispersion yaml file"""
     entries: Dict[str, Any] = {}
+    if "REFERENCES" in yml_file:
+        pass
+
+    if "COMMENTS" in yml_file:
+        pass
+    return entries
 
-    def add_table(path: str, nx_name: str, daf: pd.DataFrame):
-        disp_path = f'{path}/DISPERSION_TABLE[{nx_name}]'
 
-        if f'{path}/refractive_index' in entries:
-            entries[f'{disp_path}/refractive_index'] += daf['refractive_index'].values
+def write_dispersions(
+    yml_file: dict,
+    table_input: dispersion.TableInput,
+    model_input: dispersion.ModelInput,
+):
+    """Writes all dispersions from the dict into the entries dict"""
+
+    def write_dispersion(
+        dispersion_relation: Dict[str, Any],
+        table_input: dispersion.TableInput,
+        model_input: dispersion.ModelInput,
+    ):
+        """Writes a given dispersion relation into the entries dict"""
+
+        type_to_nx_name = {
+            "tabulated k": "dispersion_table_k",
+            "tabulated n": "dispersion_table_n",
+            "tabulated nk": "dispersion_table_nk",
+        }
+        if dispersion_relation["type"] in dispersion.SUPPORTED_TABULAR_PARSERS:
+            table_input.table_type = dispersion_relation["type"]
+            table_input.data = dispersion_relation["data"]
+            table_input.nx_name = type_to_nx_name[dispersion_relation["type"]]
+            dispersion.tabulated(table_input)
+            return
+
+        if dispersion_relation["type"] in dispersion.FORMULA_PARSERS:
+            coeffs = list(map(float, dispersion_relation["coefficients"].split()))
+            model_input.coeffs = coeffs
+            dispersion.FORMULA_PARSERS[dispersion_relation["type"]](model_input)
             return
 
-        entries[f'{path}/model_name'] = 'Table'
-        entries[f'{disp_path}/model_name'] = 'Table'
-        entries[f'{disp_path}/convention'] = 'n + ik'
-        entries[f'{disp_path}/wavelength'] = daf.index.values
-        entries[f'{disp_path}/wavelength/@units'] = 'micrometer'
-        entries[f'{disp_path}/refractive_index'] = daf['refractive_index'].values
+        raise NotImplementedError(f'No parser for type {dispersion_relation["type"]}')
```

Comment:
Consider using an if-elif-else here. We won't need the returns then. And the else can be `raise NotImple...`

Reply:
I actually like this `if...return` style. I somehow find it easier to read than if-elif-else especially for longer if-elif blocks. I can, however, change it if you think if-elif-else fits better to the general coding style in this package.

Reply:
No, it's fine. I was just mentioning it in case you would prefer that. They both read fine to me.